ui/packages/catalog-ui-search/src/main/webapp/react-component/utils/security/security.tsx

```diff
@@ -98,6 +98,18 @@ export class Security {
     return this.canAccess(user, Access.Share)
   }
 
+  isShared(): boolean {
+    return !(
+      this.res.accessGroups.length == 0 &&
+      this.res.accessGroupsRead.length == 0 &&
+      this.res.accessIndividuals.length == 0 &&
+      this.res.accessIndividualsRead.length == 0 &&
+      (this.res.accessAdministrators.length == 0 ||
```

Comment:
This is bit hard to read. Maybe break it down into smaller function calls. For example, this section could be `isTheAdministratorAlsoTheOwner` (or something similar). The other parts could be something like `areThereAnyGroupsSharing` and `areTheyAnyIndividualsSharing`.

Reply:
@glenhein I think I would find that style even harder to follow

Reply:
the current code has the benefits of repetition and symmetry

if this code is confusing, that may stem from a lack of documentation (or proper naming) for the fields in the `Restrictions` class